docs/internal-documentation/templates/specifications/tests/Template01.md

```diff
@@ -0,0 +1,262 @@
+> >
+> > The steps should state what messages (message tags) to be outputted when.
+> > Only messages with default severity level DEBUG or higher can be included.
+> >
+> > All messages with level INFO or higher must be included. I.e. the
+> > implementation should not include messages with default level INFO or higer
+> > unless included in the specification.
+> >
+> > Messages DEBUG or lower can be added in the implementation as needed.
+> >
+> > Also include statement on what other information to be accompanied
+> > with the message (included as parameter to the message tag). Example of such
+> > information is IP adresses to name servers.
+> >
+> > When refering to the data defined in **Inputs** then use the name in
+> > *italic*, e.g.:
+
+2. Create a DNSKEY query with DO flag set for *Child Zone*.
+
+> > If data sets are created, then defined them in quote marks. E.g.:
+
+4. Create three empty sets, "NSEC-Zone", "NSEC3-Zone", and
+   "No-DNSSEC-Zone".
+
+> > When used (referred to) in the steps, make the name italic. E.g.:
+
+6. Add the name server IP to the *NSEC-Zone* set (*NSEC3-Zone*
+         set).
+
+> > When a message is outputted in the steps, it is done by outputting
+> > the message tag, "T01_HAS_NSEC3" in the example below. Make the tag as a link
+> > name in italic. What the link name points at is to be defined at the bottom
+> > of the document. E.g.:
+
+8. If the NSEC (NSEC3) records do not "cover" the
+   *Non-Existent Query Name*, then output *[T01_HAS_NSEC3]*
+
+## Outcome(s)
+> > First we have standard text that should normally be the same in all
+> > test case specifications.
+
+The outcome of this Test Case is "fail" if there is at least one message
+with the severity level *[ERROR]* or *[CRITICAL]*.
+
+The outcome of this Test Case is "warning" if there is at least one message
+with the severity level *[WARNING]*, but no message with severity level
+*ERROR* or *CRITICAL*.
+
+In other cases, no message or only messages with severity level
+*[INFO]* or *[NOTICE]*, the outcome of this Test Case is "pass".
+
+> > Then there could be statements about data being outputted from the test case
+> > for use as input data for other test cases.
+
+## Special procedural requirements
+
+> > First we have standard text that should normally be the same in all
+> > test case specifications.
+
+If either IPv4 or IPv6 transport is disabled, ignore the evaluation of the
+result of any test using this transport protocol. Log a message reporting
+on the ignored result.
+
+> > Then there could be some other limitations or specialties of how or when
+> > this test case is run. E.g.:
+
+The test case is only performed if some DNSKEY record is found in the
+*Child Zone*.
+
+## Intercase dependencies
+
+> > Either the following text if there is no formal dependencies on other test
+> > cases...
+
+None.
+
+> > ...cases or specification on the outcome that this test case depend on,
```

Comment:
I think the first word "cases" can be safely removed so that the sentence starts with "or"